util/util.cpp

```diff
@@ -109,23 +148,32 @@ std::string getService(const std::string path)
     }
     catch (const sdbusplus::exception::SdBusError& ex)
     {
+        std::cerr << "Get dbus service fail. " << ex.what() << std::endl;
         return "";
     }
 
     if (response.begin() == response.end())
     {
+        std::cerr << "Sensor service not found: " << path << std::endl;
         return "";
     }
 
     return response.begin()->first;
 }
 
-void updateDbusMarginTemp(int zoneNum, int64_t marginTemp, std::string targetpath)
+void updateDbusMarginTemp(int zoneNum, double marginTemp, std::string targetpath)
 {
     auto bus = sdbusplus::bus::new_default();
-    std::string service = "xyz.openbmc_project.Hwmon.external";
+    std::string service = getService(targetpath);
 
-    dbus::SDBusPlus::setValueProperty(bus, service, targetpath, marginTemp);
+    if (service.empty())
+    {
+        std::cerr << "Sensor output path not mappable to service: " << targetpath << std::endl;
```

Comment:
Potential log flooding? Shall we refactor the code so that each sensor is wrapped up in a class, and if we get fundamental issue like this, we just remove this sensor from the polling loop?

Reply:
Perhaps good ideas for the future. This patch was already getting big, though. Adding log throttling on a per-sensor basis would be the groundwork for adding the feature of temporary sensor disqualification, as you mentioned. Disqualified sensors would be polled much less often, so if they block or do other weirdness while they are continuing to malfunction, this won't be as painful.